vnext/PropertySheets/React.Cpp.props

```diff
@@ -52,6 +42,12 @@
     <ClCompile>
       <PreprocessorDefinitions Condition="'$(ENABLE_ETW_TRACING)'=='true'">ENABLE_ETW_TRACING;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <PreprocessorDefinitions Condition="'$(ENABLE_JS_SYSTRACE_TO_ETW)'=='true'">ENABLE_JS_SYSTRACE_TO_ETW;WITH_FBSYSTRACE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+  </ItemDefinitionGroup>
+
+  <Import Project="$(ReactNativeWindowsDir)PropertySheets\JSEngine.props" />
+  <ItemDefinitionGroup>
+    <ClCompile>
       <PreprocessorDefinitions Condition="'$(UseHermes)'=='true'">USE_HERMES;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Comment:
(nit)

Should we take this opportunity to move the Hermes and V8 definitions into `JSEngine.props`?

Reply:
That would unfortunately have the side effect of leaking the preprocessor definitions to the user's project outside of MSRN. Having the separation means that React.cpp.props reads the shared property sheet about JS engine configuration to determine the preprocessor definition for MSRN specifically.

The whole property sheet structure is pretty confusing right now though in general tbh.